community/kernel/src/main/java/org/neo4j/kernel/impl/factory/EditionModule.java

```diff
@@ -96,19 +93,37 @@ protected void publishEditionInfo( UsageData sysInfo, DatabaseInfo databaseInfo,
         config.augment( singletonMap( Configuration.editionName.name(), databaseInfo.edition.toString() ) );
     }
 
-    protected AuthManager createAuthManager(Config config, LifeSupport life, LogProvider logProvider)
+    public static AuthManager createAuthManager( Config config, LogService logging )
     {
-
         boolean authEnabled = config.get( GraphDatabaseSettings.auth_enabled );
-        if ( authEnabled)
+        if ( !authEnabled )
         {
-            FileUserRepository users = life.add( new FileUserRepository( config.get( GraphDatabaseSettings.auth_store ).toPath(), logProvider ) );
-            return life.add(new BasicAuthManager( users, systemUTC(), true ));
+            return AuthManager.NO_AUTH;
         }
-        else
+
+        String key = config.get( GraphDatabaseSettings.auth_manager );
+        for ( AuthManager.Factory candidate : Service.load( AuthManager.Factory.class ) )
         {
-            return AuthManager.NO_AUTH;
+            String candidateId = candidate.getKeys().iterator().next();
+            if ( candidateId.equals( key ) )
+            {
+                return candidate.newInstance( config, logging.getUserLogProvider() );
+            }
+            else if ( key.equals( "" ) )
+            {
+                logging.getInternalLog( CommunityFacadeFactory.class )
+                        .info( "No auth manager implementation specified, defaulting to '" + candidateId + "'" );
+                return candidate.newInstance( config, logging.getUserLogProvider() );
+            }
+        }
+
+        if ( key.equals( "" ) )
+        {
+            logging.getUserLog( CommunityFacadeFactory.class )
+                    .error( "No auth manager implementation specified and no default could be loaded." );
```

Comment:
This is based on the same pattern as creating the lock manager. The default config value is the empty string "", which means that it will load the first available service. This is handy, since in community edition this will be the BasicAuthManager and in enterprise it will be the EnterpriseAuthManager. But if you want something other than the default for the edition which you are building you have the possibility to override the decision by setting the config value.
The only difference from what is done with the lock manager is what happens when no service exists at all. We decided that it was unsafe to provide a default with authentication disabled, so instead we abort and throw an exception. 
This is basically an illegal configuration of the neo4j build, so should never happen to a user. It used to be logging the error to the internal log, should I turn it back to that?